src/particles/distribution/Thermal.H

```diff
@@ -0,0 +1,360 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Ji Qiang, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_DISTRIBUTION_THERMAL
+#define IMPACTX_DISTRIBUTION_THERMAL
+
+#include <AMReX_Random.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Print.H>
+#include <ablastr/constant.H>
+#include "particles/ImpactXParticleContainer.H"
+
+#include <cmath>
+#include <cstdint>
+#include <random>
+
+namespace impactx
+{
+namespace distribution
+{
+    struct ThermalData
+    {
+        amrex::ParticleReal const tolerance = 1.0e-3; ///< tolerance for matching condition
+        amrex::ParticleReal const rin = 1.0e-10;  ///< initial r value for numerical integration
+        amrex::ParticleReal const rout = 10.0;  ///< final r value for numerical integration
+        int const nsteps = 2000;  ///< number of radial steps for numerical integration
+
+        struct Rprofile
+        {
+            amrex::ParticleReal f1;  ///< cumulative distribution of first population
+            amrex::ParticleReal f2;  ///< cumulative distribution of second population
+            amrex::ParticleReal phi1;  ///< potential generated by first population
+            amrex::ParticleReal phi2; ///< potential generated by second population
+            amrex::ParticleReal p1;  ///< normalization constant of first population
+            amrex::ParticleReal p2; ///< normalization constant of second population
+            amrex::ParticleReal rmin;  ///< minimum r value for tabulated cdf
+            amrex::ParticleReal rmax;  ///< maximum r value for tabulated cdf
+            int nbins;  ///< number of radial bins for tabulated cdf
+            amrex::ParticleReal cdf1[2001];  ///< tabulated cumulative distribution (first)
+            amrex::ParticleReal cdf2[2001];  ///< tabulated cumulative distribution (second)
+            amrex::ParticleReal Cintensity;  ///< space charge intensity parameter
+            amrex::ParticleReal bg; ///< reference value of relativistic beta*gamma
+            amrex::ParticleReal k;  ///< linear focusing strength (1/meters)
+            amrex::ParticleReal T1; ///< temperature k*T of the primary (core) population
+            amrex::ParticleReal T2; ///< temperature k*T of the secondary (halo) population
+            amrex::ParticleReal w;  ///< weight of the secondary (halo) population
+
+            Rprofile(amrex::ParticleReal kin,
+                     amrex::ParticleReal T1in,
+                     amrex::ParticleReal T2in,
+                     amrex::ParticleReal p1in,
+                     amrex::ParticleReal p2in,
+                     amrex::ParticleReal win)
```

Comment:
AMReX convention for declarations & definitions:
```suggestion
            Rprofile (amrex::ParticleReal kin,
                      amrex::ParticleReal T1in,
                      amrex::ParticleReal T2in,
                      amrex::ParticleReal p1in,
                      amrex::ParticleReal p2in,
                      amrex::ParticleReal win)
```